tests/test_synced_collection.py

```diff
@@ -0,0 +1,349 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import pytest
+import uuid
+import os
+from tempfile import TemporaryDirectory
+from collections.abc import MutableMapping
+from collections.abc import MutableSequence
+
+from signac.core.collection_api import SyncedCollection
+from signac.core.collection_api import JSONDict
+from signac.core.collection_api import JSONList
+from signac.errors import InvalidKeyError
+from signac.errors import KeyTypeError
+
+
+def testdata():
+    return str(uuid.uuid4())
+
+
+FN_DICT = 'testdict.json'
+FN_LIST = 'testlist.json'
+
+
+class TestSyncedCollectionBase():
+
+    @pytest.fixture(autouse=True)
+    def setUp(self, request):
+        self._tmp_dir = TemporaryDirectory(prefix='jsondict_')
+        request.addfinalizer(self._tmp_dir.cleanup)
+        self._fn_dict = os.path.join(self._tmp_dir.name, FN_DICT)
+        self._fn_list = os.path.join(self._tmp_dir.name, FN_LIST)
+
+    def get_testdata(self):
+        return str(uuid.uuid4())
+
+
+class TestJSONDict(TestSyncedCollectionBase):
+
+    def get_synced_dict(self, data=None):
+        ld = JSONDict(filename=self._fn_dict, data=data)
+        if data is not None:
+            ld.sync()
```

Comment:
Why is the explicit sync call necessary? This should be transparently handled by the API, right?

Reply:
Currently, if we initiate `JSONDict` with some data it modifies `self._data` according to that but, does not save until `sync` is called.